mls-rs/src/group/secret_tree.rs

```diff
@@ -83,54 +77,35 @@ impl From<Zeroizing<Vec<u8>>> for TreeSecret {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, MlsEncode, MlsDecode, MlsSize)]
-struct TreeSecretsVec(Vec<Option<SecretTreeNode>>);
-
-impl Deref for TreeSecretsVec {
-    type Target = Vec<Option<SecretTreeNode>>;
-
-    fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
-impl DerefMut for TreeSecretsVec {
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        &mut self.0
-    }
+#[derive(Clone, Debug, PartialEq, MlsEncode, MlsDecode, MlsSize, Default)]
+struct TreeSecretsVec<T: TreeIndex> {
+    #[cfg(feature = "std")]
+    inner: HashMap<T, SecretTreeNode>,
+    #[cfg(not(feature = "std"))]
+    inner: BTreeMap<T, SecretTreeNode>,
```

Comment:
I don't think this important right now, but I believe this would be the only place BTreeMap is used in the minimal configuration and thus would drive up binary size. Maybe this could be `Vec<T, SecretTreeNode>` instead if binary size becomes an issue?

Reply:
Fixed @tomleavy